src/components/CustomTextInput.tsx

```diff
@@ -0,0 +1,123 @@
+  flex: 1;
+`
+
+const ClearContainer = styled.TouchableOpacity`
+  width: ${wp('6%')}px;
+  height: ${wp('6%')}px;
+`
+
+const TextInputContainer = styled.View<{ lines: number; borderColor: string; showErrorValidation: boolean }>`
+  border: 1px solid ${props => props.borderColor};
+  padding-left: ${props => props.theme.spacings.sm};
+  padding-right: ${props => props.theme.spacings.xs};
+  border-radius: 2px;
+  height: ${props => (props.lines > 1 ? props.lines * LINE_HEIGHT : MIN_HEIGHT)}px;
+  flex-direction: row;
+  margin-bottom: ${props => (props.showErrorValidation ? 0 : props.theme.spacings.xs)};
+`
+
+const IconContainer = styled.View<{ multiLine: boolean }>`
+  align-self: ${props => (props.multiLine ? 'flex-start' : 'center')};
+  padding: ${props => props.theme.spacings.xs} 0;
+`
+
+const ErrorContainer = styled.View`
+  min-height: ${props => props.theme.spacings.lg};
+`
+
+interface CustomTextInputProps extends TextInputProps {
+  value: string
+  onChangeText: (value: string) => void
+  /** calculates the height for textArea */
+  lines?: number
+  /** shows clear indicator if set & input set */
+  clearable?: boolean
+  rightContainer?: ReactElement
+  /** Shows error message below the text input. Empty string provides a placeholder */
+  errorMessage?: string
+  /** Custom border color can be defined */
+  customBorderColor?: string
+}
+
+// depending on error and focus the border color will be defined
+const getBorderColor = (hasErrorMessage: boolean, isFocused: boolean): string => {
+  if (hasErrorMessage) {
+    return theme.colors.incorrect
+  }
+  return isFocused ? theme.colors.primary : theme.colors.textSecondary
+}
+
+const CustomTextInput: React.FC<CustomTextInputProps> = ({
+  value,
+  onChangeText,
+  lines = 1,
+  clearable = false,
+  placeholder,
+  textContentType,
+  rightContainer,
+  errorMessage,
+  editable = true,
+  onSubmitEditing,
+  customBorderColor
+}: CustomTextInputProps): ReactElement => {
+  const theme = useTheme()
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+  const hasErrorMessage = !!(errorMessage && errorMessage.length > 0)
+  const showErrorValidation = errorMessage !== undefined
+  const multiLine = lines > 1
+
+  return (
+    <>
+      <TextInputContainer
+        lines={lines}
+        borderColor={customBorderColor ?? getBorderColor(hasErrorMessage, isFocused)}
+        showErrorValidation={showErrorValidation}>
+        <StyledTextInput
+          onFocus={() => setIsFocused(true)}
+          onBlur={() => setIsFocused(false)}
+          textContentType={textContentType}
+          value={value}
+          onChangeText={onChangeText}
+          placeholder={placeholder}
+          multiline={multiLine}
+          placeholderTextColor={theme.colors.placeholder}
+          editable={editable}
+          onSubmitEditing={onSubmitEditing}
+          textAlignVertical={multiLine ? 'top' : 'center'}
+        />
+        <IconContainer multiLine={multiLine}>
```

Comment:
When adding a custom discipline there previously was no close-icon to delete the input, because at the same position is the scan-qr-code-icon. Please change, so the scan-qr-code-icon is always visible

Reply:
alright i thought maybe it would be nice to have a remove option after scanning/typing key. Because it's way more handy on mobile devices to have a single click remove option than to focus input field and remove with software keyboard. so for scanning it would just one click to remove content and then click on the scan symbol

Reply:
I think we should stick to the old behaviour, as a user cannot know that he needs to delete the input to be able to scan again.

Reply:
i removed the `clearable`